src/main/java/org/minions/devfund/damianvp/WhackAMole.java

```diff
@@ -0,0 +1,178 @@
+            this.molesLeft++;
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Method to randomly place n quantity of moles.
+     * @param moles quantity of moles to place randomly.
+     */
+    void randomPlace(int moles) {
+        Random random = new Random();
+        int max = this.moleGrid.length;
+        int min = 0;
+        boolean moleIsPlaced;
+        if (moles <= this.moleGrid.length * this.moleGrid.length) {
+            while (moles > 0) {
+                int x = random.nextInt(max - min);
+                int y = random.nextInt(max - min);
+                moleIsPlaced = place(x, y);
+                if (moleIsPlaced) {
+                    moles--;
+                }
+            }
+        }
+        System.out.println("Is not possible place more than fdsfsdf");
+    }
+
+    /**
+     * Method to whack a location in order to search a mole.
+     * @param x x location of grid game.
+     * @param y y location of grid game
+     */
+    void whack(int x, int y) {
+        if (validatedCoordinates(x, y)) {
+            if (this.moleGrid[x][y] == MOLE) {
+                System.out.println("Good, you found a mole!!");
+                this.molesLeft--;
+                this.score++;
+
+                this.trackerMoleGrid[x][y] = WHACKED;
+                this.moleGrid[x][y] = WHACKED;
+            }
+            this.attemptsLeft--;
+            verifyGameStatus();
+        }
+    }
+
+    /**
+     * Method to print the grid available for user.
+     */
+    void printGridToUser() {
+        System.out.println(Arrays.deepToString(this.trackerMoleGrid));
+    }
+
+    /**
+     * Method to print the complete grid game status.
+     */
+    void printGrid() {
+        System.out.println(Arrays.deepToString(this.moleGrid));
+    }
+
+    /**
+     * Method to validate locations.
+     * @param x x location of grid game.
+     * @param y y location of grid game.
+     * @return true, if locations are valid.
+     *         false, if location are not valid.
+     */
+    boolean validatedCoordinates(int x, int y) {
+        if (x > INVALID_RANGE && y > INVALID_RANGE && x < this.moleGrid.length && y < this.moleGrid.length) {
+            if (x == -1 && y == -1) {
+                this.attemptsLeft = 0;
+                verifyGameStatus();
+                return false;
+            }
+            return true;
+        } else {
+            System.out.println(String.format("valid coordinates should be between 0 - %s", this.moleGrid.length - 1));
+            return false;
+        }
+    }
+
+    /**
+     * Method to verify if game continue.
+     */
+    void verifyGameStatus() {
+        if (this.molesLeft == 0) {
+            System.out.println("Great you win!!!");
+            System.out.println(String.format("Score: %s", this.score));
+            return;
+        }
+        if (this.attemptsLeft <= 0) {
+            System.out.println("You Loss...  Try Again");
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this use of System.out or System.err by a logger. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)